examples/jsm/renderers/webgpu/WebGPUBackend.js

```diff
@@ -268,12 +283,45 @@ class WebGPUBackend extends Backend {
 
 	}
 
-	finishRender( renderContext ) {
+	async finishRender( renderContext ) {
 
 		const renderContextData = this.get( renderContext );
+		const occlusionQueryCount = renderContext.occlusionQueryCount;
+
+		let readBuffer;
+
+		if ( occlusionQueryCount > renderContextData.occlusionQueryIndex ) {
```

Comment:
Is this the correct condition?

Reply:
Because I think `endOcclusionQuery` is also called in the draw method?

Reply:
This should catch the case when the last queried object being rendered is the last draw call in the pass.  In the draw call. The end is inserted when a new object is seen, to allow handling of multiple draw calls per object (array cameras etc). Otherwise the endOcclusionQuery call could be at the end of the draw method.

According to the docs you can't use a querySet index in multiple beginOcclusionQuery( index ) calls in the same pass, otherwise you could just have repeated begin/end pairs around each draw() call.